Question: accessing values of variables as they are being used for provisioning an instance inside Testinfra tests

I want to use Testinfra tests to test my role.

Inside an Testinfra test I would like to access the values of variables as they are being used for provisioning the machine when `playbook.yml` is converged for some instance.
I need this since the instance's state, which I want to check, depends on the chosen values of the variables defined in role's `default/main.yml` or `vars/main.yml` file.

I tried using the following ['trick' suggested by Testinfra maintainer](https://github.com/philpep/testinfra/issues/61#issuecomment-178503626), but it only works for Ansible facts and `group_vars`/`host_vars`, not for variables defined within a role, either in `default/main.yml` or `vars/main.yml`.

Here is an example test for PostgreSQL service:
```python
def test_postgresql_running_and_enabled(Ansible, Service):
    postgresql_unit_name = Ansible("debug", "msg={{ postgresql_unit_name }}")["msg"]
    postgresql = Service(postgresql_unit_name)
    assert postgresql.is_running
    assert postgresql.is_enabled
```

Variable `postgresql_unit_name` is defined in role's `vars/main.yml`, but apparently the invocation of Ansible through Testinfra is unable to find it. Here is the error:

```python
    def test_postgresql_running_and_enabled(Ansible, Service):
        postgresql_unit_name = Ansible("debug", "msg={{ postgresql_unit_name }}")["msg"]
        postgresql = Service(postgresql_unit_name)
>       assert postgresql.is_running
E       assert <service 'postgresql_unit_name' is undefined>.is_running
```

Any ideas how I could achieve this?

--- a/src/molecule/util.py
+++ b/src/molecule/util.py
@@ -143,6 +143,7 @@ def run_command(
         else:
             env = cmd.env or env
         args = cmd.cmd
+        cwd = cmd.cwd
         stdout = cmd.stdout
         stderr = cmd.stderr
     else:

